includes/WcfceMailProcessor.inc

```diff
@@ -0,0 +1,223 @@
+
+    // For all cases related to the webform:
+    foreach ($cases as $case) {
+      // Create activity.
+      $result = civicrm_api3('Activity', 'create', [
+        'sequential' => 1,
+        'activity_type_id' => 'Email',
+        'case_id' => $case['id'],
+        'subject' => $this->message['subject'],
+        'details' => trim($this->message['body'][0]),
+        'assignee_id' => $assignee_id,
+        'source_contact_id' => $creator_id,
+      ]);
+      // Attach eml file to it.
+      if (!empty($result['values'][0]['id'])) {
+        $this->attachEmlToActivity($result['values'][0]['id'], $eml);
+      }
+    }
+
+    return $result['values'][0] ?? [];
+  }
+
+  /**
+   * Attach eml file to activity.
+   *
+   * @param int $activity_id
+   *   Activity id.
+   * @param string $eml
+   *   Contents of eml file.
+   * @param array|NULL $message
+   *   Email data. If is set to NULL then $this->message would be used.
+   *
+   * @return array|false
+   *   Attached file entity data, empty array or FALSE in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function attachEmlToActivity($activity_id, $eml, $message = NULL) {
+    if (!$activity_id) {
+      return FALSE;
+    }
+
+    $result = civicrm_api3('Attachment', 'create', [
+      'sequential' => 1,
+      'mime_type' => 'message/rfc822',
+      'entity_table' => 'civicrm_activity',
+      'entity_id' => $activity_id,
+      'content' => $eml,
+      'name' => $this->generateEmlFileName($message),
+    ]);
+
+    // Set file type to attached file.
+    // (The file_type_id db table column is always NULL and not used in civicrm
+    // at all. But we need to set some flag to differentiate original email from
+    // other attached files (and disable edit/delete for this file later), so we
+    // will use that field for this purpose and just set 1 as a value)
+    if (!empty($result['values'][0]['id'])) {
+      civicrm_api3('File', 'create', [
+        'id' => $result['values'][0]['id'],
+        'file_type_id' => 1,
+      ]);
+    }
+
+    return $result['values'][0] ?? [];
+  }
+
+  /**
+   * Generate name for eml file.
+   *
+   * @param array|NULL $message
+   *   Email data. If is set to NULL then $this->message would be used.
+   *
+   * @return string
+   */
+  public function generateEmlFileName($message = NULL) {
+    $message = $message ?? $this->message;
+    $res = date('Ymd_Hi_')
+      . preg_replace('!\W+!', '', trim(strtolower($message['subject'])))
+      . '.eml';
+
+    return $res;
+  }
+
+  /**
+   * Check if email is a case webform submission message.
+   *
+   * @return bool
+   *   FALSE if not, TRUE otherwise.
+   */
+  public function isMessageValid() {
+    $res = TRUE;
+
+    // Process webform submission emails only.
+    if ($this->message['module'] !== 'webform' || $this->message['key'] !== 'submission') {
+      $res = FALSE;
+    }
+    // Do process if a webform is configured to create/update a Case.
```

Comment:
This statement/comment sounds like we expect the elseif statement to evaluate to TRUE rather than false and should it be in an elseif condition? Probably an `If` ?

Was thinking something like this might be better?
```php
  public function isMessageValid() {
    // Process webform submission emails only.
    if ($this->message['module'] !== 'webform' || $this->message['key'] !== 'submission') {
      return FALSE;
    }
    // Don't process if a webform is not configured to create/update a Case.
    if (empty($this->message['params']['submission']->civicrm['case'])) {
      return FALSE
    }

    return TRUE;
  }

```

This way, we can do a quick return if the `message` don't meet the criteria.